### PR TITLE
CAMEL-15680: [camel-azure] Add integration test

### PR DESCRIPTION
**Pull request by luigidemasi**
- [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-azure-storage-blob/src/main/java/org/apache/camel/component/azure/storage/blob/BlobConsumer.java [hunk 1]
```diff
@@ -94,6 +94,7 @@ private Exchange createExchangeFromBlob(final String blobName, final BlobContain
 
         getEndpoint().setResponseOnExchange(response, exchange);
 
+        exchange.getIn().setHeader(BlobConstants.BLOB_NAME, blobName);
```
Comment (oscerd) [inline components/camel-azure-storage-blob/src/main/java/org/apache/camel/component/azure/storage/blob/BlobConsumer.java:97]: This doesn't seem to be related to it test

**Review by oscerd — CHANGES_REQUESTED** [review]
Just one finding.

File: components/camel-azure-storage-blob/src/main/java/org/apache/camel/component/azure/storage/blob/BlobConsumer.java (hunk 1 above)
Comment (omarsmak) [reply components/camel-azure-storage-blob/src/main/java/org/apache/camel/component/azure/storage/blob/BlobConsumer.java:97]: I guess this info can simplify the tests as I recall the name wasn't populated in the headers but now with regex and batch blob, I guess is needed to have the name in the headers, hence I don't see any problems here

**Review by omarsmak — APPROVED** [review]

**Review by oscerd — APPROVED** [review]
Sounds good the

**Review by davsclaus — APPROVED** [review]
